### PR TITLE
[v2.7] Add additional provisioning tests to OnTag

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

The OnTag runs that are ran currently lack a number of different provisioning tests. These include node provisioning tests for RKE2/K3s and custom cluster provisioning tests for RKE2/K3s. This PR aims to add support for those missing provisioning tests so that they are covered in the OnTag runs.
 
## Solution

Added EC2/Azure node provider support for RKE2/K3s in the `Jenkinsfile_ontag_matrix` file. In the same file, added support for custom cluster provisioning for RKE2/K3s.

Additionally, this PR changes AKS version to get the second highest version available. In Azure Portal, the highest version available typically is not the most stable version. This PR fixes that concern.
 
### Automated Testing

Jenkins runs will be provided to the reviewers.

### CODE REVIEW

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (slickwarren) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:191]: could you update this to the latest patch while you're here (or 1.25 if PSA won't mess anything up)?

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (slickwarren) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:283]: The reason for 2 servers was also to distribute load; depending on the outcome of the jenkins run, we should consider moving half of the new tests to server 2.

File: tests/validation/tests/v3_api/scripts/provision-config.yaml
Comment (slickwarren) [inline tests/validation/tests/v3_api/scripts/provision-config.yaml:1]: it would be nice to override or add to this from a jenkins argument (in the jenkins-job-builder side of things) but given the nature of this jenkinsfile, I understand that may not be possible as there are already other args that must be hard coded like this.

**Review by slickwarren — COMMENTED** [review]
just some nits, if you could fix them with the jenkinsfile updates that would be great. otherwise LGTM

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:283]: Understood, thanks for that. I split them, 3 into each server, respectively.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (igomez06) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:282]: Also these jobs are just provisioning the clusters correct? When are the actual tests being ran against them?

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (igomez06) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: If these jobs were just provisioning, why would you need all of these env vars?

**Review by igomez06 — COMMENTED** [review]
How it is, I am very certain it won't work. Could we sync up, and I can make a temp job for you to test in Jenkins? I don't want to merge anything that will break this job, since we use it for release testing.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (igomez06) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:282]: The certification test provision the cluster then run certain tests against the clusters.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (igomez06) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:283]: Yeah I don't want to merge anything that still needs to be tested, let's sync up so I can make temp job for you test these changes.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: I had these params so when I called the Go jobs, I just reference the specific params.

Comment (igomez06) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: I still don't quite understand. Is the goal to run the certification tests against the new provisioned clusters? Also if you are just running the go freeform job, these params are not necessary as the freeform job only needs the config.

Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: So the goal here is to essentially extend the coverage of the tests that the certification tests already do. We want to have RKE2/K3s run on EC2/Azure/custom clusters. So we decided using Go would be best as we have the freeform jobs to more so easily run them.

I can remove the params in that case and just have the config, I initially thought I needed all of the params in this ontag file.

Comment (igomez06) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: So we will have the same certification tests, but in Go? The part I'm missing is where the certification tests come into play. ie these cert jobs all run these tests against the cluster `[ "test_wl", "test_dns_record", "test_rbac","test_connectivity", "test_ingress",
                        "test_secrets", "test_registry", "test_service_discovery", "test_websocket"]`

So what you are setting is the params for the job you want to run, since the freeform job only takes the config file, the rest really won't do anything.

Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: Sorry, I'm not explaining it well. To clarify, we're not creating the same certification tests in Go. The end goal is to add new tests as mentioned in the original [task](https://github.com/rancher/qa-tasks/issues/429#issue-1283170650) that supplement the existing certification tests. The Go test additions would not be running certification tests, they are strictly just provisioning tests to add more coverage to the daily ontag runs.

Comment (igomez06) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:220]: Got it thank you!! So yeah the other params are not necessary only the params that the go freeform job needs.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (slickwarren) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:89]: just double checking, do you not need the ARM credentials too?

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (slickwarren) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:117]: looks like you have some whitespace here?

File: tests/validation/tests/v3_api/test_aks_cluster.py
Comment (slickwarren) [inline tests/validation/tests/v3_api/test_aks_cluster.py:63]: would you mind adding a comment as to why the order is what it is / what these are?

**Review by slickwarren — COMMENTED** [review]
some small nits, overall LGTM; pending jenkins run?

File: tests/validation/tests/v3_api/test_aks_cluster.py
Comment (markusewalker) [reply tests/validation/tests/v3_api/test_aks_cluster.py:63]: Sure, I'll go ahead and do that.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:117]: Nice catch! I'll fix that.

**Review by slickwarren — APPROVED** [review]
LGTM

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (slickwarren) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:160]: we're getting a security warning I think due to these additions. 
```
A secret was passed to "build" using Groovy String interpolation, which is insecure.
 		 Affected argument(s) used the following variable(s): [AWS_ACCESS_KEY_ID, AZURE_SUBSCRIPTION_ID, AZURE_CLIENT_SECRET, AWS_SECRET_ACCESS_KEY]
```

Can you follow the guidelines here to prevent the insecurity? https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#string-interpolation

**Review by slickwarren — CHANGES_REQUESTED** [review]
there's some marked insecurities in the jenkins runs, we should fix these before merging.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (igomez06) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:218]: So I don't know if you forgot to update, but `CATTLE_TEST_URL` and `ADMIN_TOKEN` are not needed for the go job

**Review by igomez06 — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:160]: Good catch, looks like it's because I used double quotations and not single quotes, I'll make that change

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix:218]: I can remove those

Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix]: Resolved.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix
Comment (markusewalker) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_ontag_matrix]: Resolved.

**Review by slickwarren — APPROVED** [review]
lgtm

**Review by igomez06 — APPROVED** [review]
looks good

**Review by slickwarren — APPROVED** [review]

**Pull request merged**
